src/plugin/featureaction/featureactionmanager.js

```diff
@@ -114,6 +114,14 @@ plugin.im.action.feature.Manager.prototype.initialize = function() {
   for (var i = 0, n = sources.length; i < n; i++) {
     this.addSource_(sources[i]);
   }
+
+  var importActionMan = os.im.action.ImportActionManager.getInstance();
+  var actionEntries = importActionMan.getActionEntries();
+  for (var i = 0, n = actionEntries.length; i < n; i++) {
+    if (dm.getDescriptor(actionEntries[i].getType()) == null) {
```

Comment:
This block is deleting all feature actions when I refresh the app, even those made on a layer with a descriptor.

Reply:
This is not what happens for me. It only deletes feature actions on merged or joined layers.

Reply:
I'm still seeing the behavior. To reproduce:

- Add the [GDP Geoserver](https://gdp-geoserver.dev.dev.east.paas.geointservices.io/geoserver/ows).
- Turn on the VIIRS layer.
- Import [VIIRS Detection FAs.xml.txt](https://github.com/ngageoint/opensphere/files/3216519/VIIRS.Detection.FAs.xml.txt) (drop the `.txt` extension, GitHub won't upload `.xml`).
- Right click the layer > Most Recent, then load some data. You should see purple stars from the FA.
- Reload the app. The FA should be gone.

Reply:
This happens because the OGC descriptor has aliases (alternate ID's) that get populated when it finishes loading, and ties it to the WMS/WFS layers. When this check runs, the server is still in a loading state so those aliases have not been populated and the descriptor cannot be found. It's also possible the provider is disabled and will not load, so we can't simply wait for all providers to finish loading.

Instead of trying to clean up the entries when restoring the state, we can prevent it from being saved in the first place. The parent class (`os.filter.FilterEntry`) has a temporary flag set with `setTemporary` that will prevent `os.im.action.ImportActionManager#save` from saving the feature action.

I would suggest setting the flag in `os.ui.filter.ui.EditFiltersCtrl#finish`, which will prevent both feature actions and filters from being persisted to storage if the entry does not have a descriptor.